app/controllers/evidence_controller.rb

```diff
@@ -154,17 +154,18 @@ def set_or_initialize_evidence
     end
   end
 
-  # If the user selects "Add new issue" in the Evidence editor, we create an empty skeleton
-  def create_issue
-    Issue.create do |issue|
-      issue.text = "#[Title]#\nNew issue auto-created for node [#{@node.label}]."
-      issue.node = current_project.issue_library
-      issue.author = current_user.email
+  def evidence_params
+    evidence_params = params.require(:evidence).permit(:author, :content, :issue_id, :node_id)
+
+    if evidence_params[:issue_id].blank?
```

Comment:
@sean-yeoh this behaviour is distinct from `evidence_params` and should be extracted to a different method. In fact, `evidence_params` shouldn't really concern itself with the :issue_id contents. You can do this in the action, or in a filter, but it doesn't seem to fit here.

In fact, I still think it belongs in a class method in Issue.

```
evidence.issue = Issue.autogenerate_from(evidence_params) unless evidence_params[:issue_id]
```

(or `.create_from`, etc.)

or whatever the right logic is.